packages/dashboard/components/consent/utils/consentTransactionUtils.ts

```
@@ -148,29 +150,36 @@ export const addPermittedActions = async (
   });
   await litContracts.connect();
 
-  for (const ipfsCid of toolIpfsCids) {
-    try {
-      // Check if this action is already permitted
-      const isAlreadyPermitted = await litContracts.pkpPermissionsContractUtils.read.isPermittedAction(
-        agentPKPTokenId,
-        ipfsCid
-      );
-
-      // Handle special case for DCA
-      if (ipfsCid === process.env.NEXT_PUBLIC_DCA_TOOL_IPFS_CID) {
+  for (const ipfsCid of policyIpfsCids) {
+    if (IPFS_POLICIES_THAT_NEED_SIGNING[ipfsCid]) {
+      try {
         const isPolicyPermitted = await litContracts.pkpPermissionsContractUtils.read.isPermittedAction(
           agentPKPTokenId,
-          process.env.NEXT_PUBLIC_DCA_POLICY_IPFS_CID!
+          ipfsCid
         );
+
         if (!isPolicyPermitted) {
-          console.log(`Adding DCA policy for ${process.env.NEXT_PUBLIC_DCA_POLICY_IPFS_CID}`);
-          await litContracts.addPermittedAction({
-            ipfsId: process.env.NEXT_PUBLIC_DCA_POLICY_IPFS_CID!,
+          console.log(`Adding sign permission for policy ${ipfsCid}, ${IPFS_POLICIES_THAT_NEED_SIGNING[ipfsCid].description}`);
+          const tx = await litContracts.addPermittedAction({
+            ipfsId: ipfsCid,
             pkpTokenId: agentPKPTokenId,
             authMethodScopes: [AUTH_METHOD_SCOPE.SignAnything],
           });
+          console.log(`Added sign permission for policy ${ipfsCid} - Transaction hash: ${tx.transactionHash}`);
         }
+      } catch (error) {
+        console.error(`Error adding DCA policy permission for ${ipfsCid}:`, error);
```

We can't just ignore failures in this flow to check for or add new tool policy permissions -- if we do this, the user will appear to sign up successfully, we'll provide a JWT to the app via redirect, _but when the app tries to actually execute tools for the user, they will fail due to invalid delegations_.

Consider that, in the scenario I just described, _the only resolution the app can provide to the user_ would be for the user to return _back to this screen_ and go through _this process again_ to fix the problem. Then, consider that _it might fail again_, but if so, this code will once again ignore the failure, and the user will end up 'signed in' to the app, but their ability to actually use the app will still be broken.

We need to track an error state of some kind, and make the consent page render that error state if it is set -- that wya, If this logic throws an error,  the consent page can tell the user there was an error and give them a 'retry' button or something so they can re-run this code from the top-down, rather than us just ignoring the error and sending them off to an app that we can reasonably assume (right here in this code) will be in a broken state.